app/parsing_scripts/git_update.py

```diff
@@ -73,21 +72,19 @@ def get_list_of_changed_ports(new_hash=False, old_hash=False, root=BASE_DIR):
             for line in s:
                 portname = line.split('/')[1]
                 if portname not in updated_ports:
-                    updated_ports.append(portname)
+                    updated_ports.append(portname.lower())
```

Comment:
I did this just to avoid any missed ports because of case. Port names are case insensitive and at some places like `dependencies`, some maintainers have not been consistent with case. So, I wanted to avoid that situation here. Could we improve this in some other way?

Reply:
First: use dictionary rather than list, else you'll end up in quadratic time (with 30.000 ports and all of them updated, you can easily generate a billion string comparisons).

Second (even more important): this command gives you PATHS!!! rather than ports that need updating. If one changes `perl5`, any of its subports may be affected (`perl5.30`, `perl5.28`, ...), not just the main one.

Important:
* You need an unit test to check what happens when directory is renamed from lowercase to uppercase.
* You need an unit test to ensure that subports all get updated / removed / obsoleted etc. when something changes.